include/ideep/operators/conv.hpp

```diff
@@ -1258,11 +1271,18 @@ struct convolution_forward
       const lowp_kind alowp_kind = u8s8,
       const engine& aengine = engine::cpu_engine()) {
     bool is_channels_last = src.get_desc().is_channels_last() || weights.get_desc().is_channels_last();
+    bool is_fp32 = src_scales.empty() && weights_scales.empty() && dst_scales.empty();
     static tensor dummy_bias;
-    do_prepare</*with_bias=*/false>(
-        param, src, weights, dummy_bias, dst_dims, dst, strides, dilates,
-        padding_l, padding_r, groups, src_scales, weights_scales, dst_scales,
-        zero_point_t(), zero_point_t(), is_channels_last, attr, aalgorithm, aprop_kind, alowp_kind, aengine);
+    if (is_fp32) {
+      do_prepare</*with_bias=*/false>(
+          param, src, weights, dummy_bias, dst_dims, dst, strides, dilates,
+          padding_l, padding_r, groups, is_channels_last, attr, aalgorithm, aprop_kind, aengine);
+    } else {
+      do_prepare</*with_bias=*/false>(
+          param, src, weights, dummy_bias, dst_dims, dst, strides, dilates,
+          padding_l, padding_r, groups, src_scales, weights_scales, dst_scales,
+          zero_point_t(), zero_point_t(), is_channels_last, attr, aalgorithm, aprop_kind, alowp_kind, aengine);
+    }
   }
 
   // DEPRECATED
```

Comment:
Can we add a comment in the function spec that weights is expected to be reordered before calling into this function?

Reply:
Sure